.github/README.md

```diff
@@ -0,0 +1,392 @@
+* Workflow will: 
+  - Build the code pushed in each of the Build Group images. 
+  - Package the built code and upload a tarball to the *production* S3 bucket. 
+  - Create all Target Group images and install the prebuilt tarball on each.
+  - Publish an NPM package upon successful completion of all steps above. When version tag is `alpha`, package will be NPM tagged same. When it is a release version, package will be NPM tagged `latest`.
+* Workflow ensures node-pre-gyp setup is working in *production* for a wide variety of potential customer configurations.
+* Workflow publishing to NPM registry exposes the NPM package (and the prebuilt tarballs in the *production* S3 bucket) to the public.
+* Note: @appoptics/apm-bindings is not meant to be directly consumed. It is developed as a dependency of [appoptics-apm](https://www.npmjs.com/package/appoptics-apm).
+
+```
+push semver tag ─► ┌────────────────────────────┐ ─► ─► ─►
+push alpha tag     │Build Group Build & Package │ S3 Package
+                   └┬───────────────────────────┘ Production
+                    │
+                    │   ┌────────────────────┐     │
+                    └─► │Target Group Install│ ◄── ▼
+                        └┬───────────────────┘
+                         │
+                         │   ┌───────────┐
+                         └─► │NPM Publish│
+                             └───────────┘
+```
+## Maintenance
+
+> **tl;dr** There is no need to modify workflows. All data used is externalized.
+
+### Definitions
+* Local images are defined in [docker-node](./docker-node).
+* S3 Staging bucket is defined in [package.json](../package.json).
+* S3 Production bucket is defined in [package.json](../package.json).
+* [Build Group](./config/build-group.json) are images on which the various versions of the add-on are built. They include combinations to support different Node versions and libc implementations. Generally build is done with the lowest versions of the OSes supported, so that `glibc`/`musl` versions are the oldest/most compatible.
+* [Fallback Group](./config/fallback-group.json) images include OS and Node version combinations that *can* build for source.
+* [Prebuilt Group](./config/prebuilt-group.json) images include OS and Node version combinations that *can not* build for source and thus require a prebuilt tarball.
+* [Target Group](./config/target-group.json) images include a wide variety of OS and Node version combinations. Group includes both images that can build from code as well as those which can not.
+
+### Adding an image to GitHub Container Registry
+
+1. Create a docker file with a unique name to be used as a tag. Common is to use: `{node-version}-{os-name-version}` (e.g `16-ubuntu20.04.2.Dockerfile`). If image is a build image suffix with `-build`.
+2. Place a Docker file in the `docker-node` directory.
+3. Due to [GitHub Action issues](https://github.com/actions/runner/issues/1202) Manual trigger of [docker-node.yml] is required.
```

Comment:
It looks like the `docker-node.yml` workflow is getting [automatically triggered on push](https://github.com/appoptics/appoptics-bindings-node/actions/workflows/docker-node.yml?query=event%3Apush), so point 3 is outdated?

Reply:
Edits after merge.